homeassistant/components/onvif/camera.py

```diff
@@ -85,20 +93,40 @@ def handle_ptz(service):
             target_cameras = [camera for camera in all_cameras
                               if camera.entity_id in entity_ids]
         for camera in target_cameras:
-            camera.perform_ptz(pan, tilt, zoom)
+            await camera.async_perform_ptz(pan, tilt, zoom)
+
+    hass.services.async_register(DOMAIN, SERVICE_PTZ, async_handle_ptz,
+                                 schema=SERVICE_PTZ_SCHEMA)
+
+    _LOGGER.debug("Constructing the ONVIFHassCamera")
+
+    hassCamera = ONVIFHassCamera(hass, config)
 
-    hass.services.register(DOMAIN, SERVICE_PTZ, handle_ptz,
-                           schema=SERVICE_PTZ_SCHEMA)
-    add_entities([ONVIFHassCamera(hass, config)])
+    await hassCamera.async_initialize()
+
+    async_add_entities([hassCamera])
+    return True
 
 
 class ONVIFHassCamera(Camera):
     """An implementation of an ONVIF camera."""
 
     def __init__(self, hass, config):
-        """Initialize a ONVIF camera."""
+        """Initialize an ONVIF camera."""
         super().__init__()
+
+        _LOGGER.debug("Importing dependencies")
+
         import onvif
+        import zeep
+        from onvif import ONVIFCamera
+
+        # Note: important imports foor zeep and onvif-zeep
+        def zeep_pythonvalue(self, xmlvalue):
+            return xmlvalue
+        zeep.xsd.simple.AnySimpleType.pythonvalue = zeep_pythonvalue
```

Comment:
lint: If this is required, you will need to add a lint disable above this line `# pylint: disable=no-member`